Components/webComponents/cartComponents/CartItem.js

```diff
@@ -38,27 +38,44 @@ export class CartItem extends HTMLElement {
         this.querySelector(".cart-delete-item").addEventListener(
             "click",
             function () {
-                publishEvent("removeItem", this.productId);
+                this.dispatchEvent(
+                  new CustomEvent("onItemRemoved", {
+                      detail: this.productId,
+                      bubbles: true
+                  })
+                );
             }.bind(this)
         );
 
         this.querySelector(".cart-item-add-button").addEventListener(
             "click",
             function () {
-                publishEvent("quantityUpdated", {
-                    productId: this.productId,
-                    quantity: this.quantity + 1,
-                });
+                this.dispatchEvent(
+                  new CustomEvent("onQuantityUpdated", {
+                      detail: {
+                          productId: this.productId,
+                          quantity: this.quantity + 1,
+                      },
+                      bubbles: true
+                  })
+                );
             }.bind(this)
```

Comment:
Why do we need to add `.bind(this)` at the end of the event handling functions?

Reply:
And is there a way we can get rid of the `bind`s?

Reply:
using bind(this) passes the values attatched as parameters to the scope of the class in this case.

to avoid using it we can use ES6 and just write our function as so. `() =>`